Snowflake.Data.Tests/UnitTests/SFFileTransferAgentTests.cs

```diff
@@ -0,0 +1,551 @@
+        CancellationToken _cancellationToken;
+
+        // Mock response data
+        PutGetResponseData _responseData;
+        SFFileTransferAgent _fileTransferAgent;
+        SFSession _session;
+
+        // Mock PUT/GET queries
+        string _putQuery;
+        const string GetQuery = "GET @DB.SCHEMA.%MOCKTABLE file://;";
+
+        // Mock file content
+        const string FileContent = "FTAFileContent";
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            // Base object's names on worker thread id
+            var threadSuffix = TestContext.CurrentContext.WorkerId?.Replace('#', '_');
+
+            // Set values for thread variables
+            t_realSourceFilePath = $"realSrcFilePath_{threadSuffix}.txt";
+            t_localLocation = $"mockLocalLocation_{threadSuffix}";
+            t_locationStage = $"mock-customer-stage_{threadSuffix}";
+
+            // Set values for members that depend on thread variables
+            _srcLocations = new List<string>()
+            {
+                t_realSourceFilePath
+            };
+            _putQuery = "PUT file://" + t_realSourceFilePath + " @DB.SCHEMA.%MOCKTABLE;";
+            _location = Path.GetFullPath(t_locationStage + "/" + LocationId + "/" + LocationPath);
+
+            _responseData = new PutGetResponseData()
+            {
+                autoCompress = false,
+                encryptionMaterial = _encryptionMaterial,
+                localLocation = t_localLocation,
+                overwrite = false,
+                parallel = Parallel,
+                presignedUrl = null,
+                presignedUrls = null,
+                queryId = null,
+                rowSet = null,
+                rowType = null,
+                sourceCompression = AutoDetect,
+                sqlState = null,
+                src_locations = _srcLocations,
+                stageInfo = new PutGetStageInfo()
+                {
+                    location = _location,
+                    locationType = SFRemoteStorageUtil.LOCAL_FS, // Use local storage for testing
+                    path = LocationPath,
+                    presignedUrl = null,
+                    stageCredentials = null
+                },
+                statementTypeId = 0,
+                threshold = 209715200 // Server default threshold
+            };
+
+            _cancellationToken = new CancellationToken();
+
+            _session = new SFSession(ConnectionStringMock, null);
+
+            // Upload setup
+            // Write mock file to upload
+            File.WriteAllText(_srcLocations[0], FileContent);
+            _sourceFileSize = new FileInfo(_srcLocations[0]).Length;
+
+            // Download setup
+            // Write mock file in the local location to download
+            if (!Directory.Exists(_location))
+            {
+                Directory.CreateDirectory(_location);
+            }
+            File.WriteAllText(_location + t_realSourceFilePath, FileContent);
```

Comment:
As I understand these steps aren't needed for Upload tests. Maybe split somehow Upload and Download setup so that there is no unnecessary writing on the disk.

Reply:
Sounds good, will do